src/api/MainSolver.cc

```diff
@@ -280,7 +283,7 @@ sstat MainSolver::giveToSolver(PTRef root, FrameId push_id) {
     ClauseCallBack callBack;
     ts.setClauseCallBack(&callBack);
     ts.Cnfizer::cnfize(root, push_id);
-    bool keepPartitionsSeparate = getConfig().produce_inter();
+    bool keepPartitionsSeparate = getConfig().produce_proof();
```

Comment:
```suggestion
    bool trackPartitions = getConfig().produce_proof();
```

I think it would be better to have a method `bool trackPartitions() const`, same as in `Theory`. (maybe one can just use the other?)

It can either check `SMTConfig::produce_proof()` or `SMTConfig::produce_interpolants()` (and later also `SMTConfig::produce_unsat_core()`).
I think the first option is slightly better, because then the proof is a bit closer to the original formula than in the second case.